src/states/adult.rs

```diff
@@ -64,17 +66,19 @@ pub struct Adult {
     peer_map: PeerMap,
     peer_mgr: PeerManager,
     poke_timer_token: u64,
+    add_timer_token: u64,
     routing_msg_filter: RoutingMessageFilter,
     timer: Timer,
 }
 
 impl Adult {
-    pub fn from_proving_node(
+    pub fn from_joining_peer(
```

Comment:
It can also be from `Edler` as well. Elder and Adult can transition between each other in cases of Elder relocate. Also a joining peer could become an Elder immediately if the `Elder_size` < section_size()

Reply:
When we actually allow the sections to keep non-promoted Adults, yes, but for now we don't have this mechanism in place - currently all Adults are expected to be promoted to Elders shortly after joining, and there is no possibility of going the other way. But when we implement that, it's possible that the function can be made to serve both purposes :+1:

Reply:
The mechanics are in place with Adams the last task. But I am OK we address this when we couple that all together in the node join task. :+1: